### PR TITLE
be: change bitbucket dpu api call return type from `RepoIdentifier[]` to `BitbucketDBRepo[]`

### PR DESCRIPTION
**Pull request by MuskanPaliwal**

### CODE REVIEW

File: utils/db/repos.ts [hunk 1]
```diff
@@ -243,3 +262,55 @@ export const removeRepoConfigForInstallIdForOwner = async (installId: string, re
...
+			repo_obj->>'name' AS repo_name,
+			repo_obj->'workspace'->>'slug' AS repo_owner,
+			'bitbucket' AS repo_provider,
+			repo_obj->'workspace'->>'slug' AS workspace,
+			(SELECT clone->>'href' 
+			FROM jsonb_array_elements(repo_obj->'links'->'clone') AS clone 
+			WHERE clone->>'name' = 'ssh') AS clone_ssh_url,
+			(repo_obj->>'is_private')::BOOLEAN AS is_private,
+			repo_obj->'project' AS project
+			jsonb_set('{}', '{uuid}', to_jsonb(repo_obj->>'uuid')) AS metadata
```
Comment (tapishr) [inline utils/db/repos.ts:288]: json can be used instead of jsonb

File: utils/providerAPI/getUserRepositories.ts [hunk 2]
```diff
@@ -56,6 +57,12 @@ type BitbucketRepoObj = {
 	uuid: string,
 }
 
+type BitbucketReposResult = {
+    bitbucketReposObjs: BitbucketRepoObj[];
+    repoIdentifiers: RepoIdentifier[];
+};
+
+
```
Comment (MuskanPaliwal) [inline utils/providerAPI/getUserRepositories.ts]: same as above?!

File: utils/providerAPI/getUserRepositories.ts [hunk 3]
```diff
@@ -136,21 +143,23 @@ const getUserRepositoriesForGitHub = async (access_token: string, authId?: strin
...
 }
 
-export const getUserRepositoriesForBitbucket = async (access_key: string, authId?: string): Promise<RepoIdentifier[]> => {
+export const getUserRepositoriesForBitbucket = async (access_key: string, authId?: string): Promise<BitbucketReposResult> => {
 	const workspacesData = await Bitbucket.retrieveAllPages<BitbucketWorkspaceObj>(`/user/permissions/workspaces`, access_key, authId);
 	const workspaces = workspacesData.map((workspaceObj) => workspaceObj.workspace.slug);
 
-	const repositories: RepoIdentifier[] = [];
+	const bitbucketRepos: BitbucketRepoObj[] = [];
+    const repoIdentifiers: RepoIdentifier[] = [];
```
Comment (MuskanPaliwal) [inline utils/providerAPI/getUserRepositories.ts]: indentation?

**Review by MuskanPaliwal — COMMENTED** [review]
Looks good to me. @tapishr please have a look once.

**Review by tapishr — APPROVED** [review]

**Pull request merged**